v1.3.0 release notes (#5900)
spiceai/spiceai: docs/release_notes/v1.3.0.md

- **New range() Table Function**: A new table-valued function `range(start, stop, step)` has been added to make it easy to generate integer sequences — similar to PostgreSQL’s generate_series() or Spark’s range(). Example: `SELECT * FROM range(1, 10, 2);`

- **UNION [ALL | DISTINCT] BY NAME Support**: DataFusion now supports `UNION BY NAME` and `UNION ALL BY NAME`, which align columns by name instead of position. This matches functionality found in systems like Spark and DuckDB and simplifies combining heterogeneously ordered result sets.

  Example:

  ```sql
  SELECT col1, col2 FROM t1
  UNION ALL BY NAME
  SELECT col2, col1 FROM t2;
  ```

See the [DataFusion 46.0.0 release notes](https://datafusion.apache.org/blog/2025/03/24/datafusion-46.0.0/) for details.

Spice.ai upgrades to the latest minus one DataFusion release to ensure adequate testing and stability. The next upgrade to DataFusion v47 is planned for Spice v1.4.0 in June.

## What's New in v1.3.0

> - **Cached Query Engine Performance**: The Spice query engine delivers up to 18% higher throughput and 20% lower latency for cached queries.

Reviewer: ````suggestion
- **Improved Results Cache Performance & Hashing Algorithm:** In addition to performance optimizations of up to 18% higher throughput and 20% lower latency for results caching, the Runtime now supports specifying an alternative hashing algorithm, `ahash`. The `siphash` algorithm is used by default, which can be changed by specifying a Runtime parameter:

```yaml
runtime:
  results_cache:
    hashing_algorithm: ahash
```

Using `ahash` as the cache key hashing algorithm results in improved performance, particularly for large queries or query plans. In addition to using `ahash`, the results cache performance optimizations have improved the 99th percentile request latency and total requests/second for queries with large query results (100k+ cached rows). These charts displays the 99th percentile request latency and total requests/second tested against the [TPCH Query #17](https://github.com/spiceai/spiceai/blob/trunk/crates/test-framework/src/queries/tpch/q17.sql) on a scale factor 5 dataset (30+ million rows, 5gb dataset size):

| Latency | Req/sec |
| - | - |
| ![Improvements for the 99th percentile query latency, compared against 1.2.2 with cache key type and hashing algorithm.](https://github.com/user-attachments/assets/4cfedeaf-df58-403e-b861-e0eac436ff00) | ![Improvements for the requests/second, compared against 1.2.2 with cache key type and hashing algorithm.](https://github.com/user-attachments/assets/0bc3f866-82cf-4f41-88ec-52283c084675) |

`ahash` is excluded from performance comparisons in 1.2.2, because the hashing algorithm was not available in this release.

To learn more, refer to the [Results Cache Hashing Algorithm documentation](https://spiceai.org/docs/reference/spicepod/runtime#runtimehashing_algorithm).
````

Release notes from https://github.com/spiceai/spiceai/issues/5832.

- [@Jeadie](https://github.com/Jeadie)
- [@phillipleblanc](https://github.com/phillipleblanc)
- [@Sevenannn](https://github.com/Sevenannn)
- [@lukekim](https://github.com/lukekim)
- [@kczimm](https://github.com/kczimm)
- [@ewgenius](https://github.com/ewgenius)
- [@peasee](https://github.com/peasee)
- [@sgrebnov](https://github.com/sgrebnov)

## Breaking Changes

No breaking changes.

## Cookbook Updates

- Added [Accelerated Views](https://github.com/spiceai/cookbook/blob/trunk/views/README.md): Pre-calculate and materialize data derived from one or more underlying datasets.

The [Spice Cookbook](https://spiceai.org/cookbook) now includes 67 recipes to help you get started with Spice quickly and easily.

## Upgrading

To upgrade to v1.3.0, use one of the following methods:

**CLI**:

```console
spice upgrade
```

**Homebrew**:

```console
brew upgrade spiceai/spiceai/spice
```

**Docker**:

Pull the `spiceai/spiceai:1.3.0` image:

```console
docker pull spiceai/spiceai:1.3.0
```

For available tags, see [DockerHub](https://hub.docker.com/r/spiceai/spiceai/tags).

**Helm**:

```console
helm repo update
helm upgrade spiceai spiceai/spiceai
```

## What's Changed

## Dependencies

> - **DataFusion**: [upgraded to v46]((https://datafusion.apache.org/blog/2025/03/24/datafusion-46.0.0/))

Reviewer 2: ```suggestion
- **DataFusion**: [Upgraded to v46](https://datafusion.apache.org/blog/2025/03/24/datafusion-46.0.0/)
```

> - **[DataFusion](https://datafusion.apache.org/blog/2025/03/24/datafusion-46.0.0/):** upgraded to v46.
> - **[Apache Arrow](https://github.com/apache/arrow-rs/releases/tag/54.3.0):** Upgraded to v54.3.0.
> - **[delta_kernel](https://crates.io/crates/delta_kernel/0.10.0):** Upgraded to v0.10.0.

Reviewer 2: ```suggestion
- **DataFusion**: [Upgraded to v46](https://datafusion.apache.org/blog/2025/03/24/datafusion-46.0.0/)
- **Apache Arrow**: [Upgraded to v54.3.0](https://github.com/apache/arrow-rs/releases/tag/54.3.0)
- **delta_kernel**: [Upgraded to v0.10.0](https://crates.io/crates/delta_kernel/0.10.0)
```
